modules/stochastic_tools/src/surrogates/GaussianProcess.C

```diff
@@ -0,0 +1,79 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GaussianProcess.h"
+#include "Sampler.h"
+
+registerMooseObject("StochasticToolsApp", GaussianProcess);
+
+InputParameters
+GaussianProcess::validParams()
+{
+  InputParameters params = SurrogateModel::validParams();
+  params.addClassDescription("Computes and evaluates Gaussian Process surrogate model.");
+  return params;
+}
+
+GaussianProcess::GaussianProcess(const InputParameters & parameters)
+  : SurrogateModel(parameters),
+    _sample_points(getModelData<std::vector<std::vector<Real>>>("_sample_points")),
+    _training_params(getModelData<RealEigenMatrix>("_training_params")),
+    _training_params_mean(getModelData<RealEigenVector>("_training_params_mean")),
+    _training_params_var(getModelData<RealEigenVector>("_training_params_var")),
+    _training_data(getModelData<RealEigenMatrix>("_training_data")),
+    _training_data_mean(getModelData<RealEigenVector>("_training_data_mean")),
+    _training_data_var(getModelData<RealEigenVector>("_training_data_var")),
```

Comment:
I agree there should be a class of some form to handle this.

However I think it should maybe some form of general utility rather than exclusive to GP. Standardizing your training data/params is seemingly a very common thing in ML and adjacent fields, so it would be nice to have some small object that handles this in general sense.


If we did go down this path, there is a design decision if the object should:

     1. Store mean, variance, as well as the **data** used to generate those variances. In this case the "standardizer" is in some way tied to the data that was used to generate it. In this case the above 6 data loads would be condensed to 2, as I would just need to pass 2 standardizer objects: one for params and one for data.

     2: Only store the mean and variance, which is enough to apply/un-apply the standardization, but would not retain the data used to generate those values. In some ways this feels more flexible as it doesn't distinguish between the data that generated it and data it is begin applied to. It also allows the user to directly set the mean/var without a dataset. This is how I have seen similar object implemented in other codes.  In this scenario I would still need to data load 4 objects. 2 standardizers as well as the 2 arrays corresponding to the raw data/params.

Reply:
Ok, this was done using the Standardizer utility object. Design wise it uses #2. Its pretty bare bones at the moment but does its job for this implementation. As our data-science efforts further expand, the standardizer object should undergo further development to:

1. Support more data types for the input/output: Namely various matrix classes, vectors, and arrays in moose. Maybe templated, maybe just some overrides, decisions will need to be made.

2. Support distributed data. At the moment the GP design is not set up for distributed data, so neither is the standardizer at the moment. This will absolutely be needed if/when the standardizer is used in a distributed trainer OR GP is updated to support parallelization.